packages/vega-scenegraph/test/canvas-offscreen-test.js

```diff
@@ -0,0 +1,120 @@
+import tape from 'tape';
+import {CanvasRenderer as Renderer, sceneFromJSON} from '../index.js';
+import './__init__.js';
+
+// Mock OffscreenCanvas for Node.js testing environment
+class MockOffscreenCanvas {
+  constructor(width, height) {
+    this.width = width;
+    this.height = height;
+    this._context = null;
+  }
+
+  getContext(type) {
+    if (type === '2d' && !this._context) {
+      // Create a minimal mock 2D context
```

Comment:
Can we not use node canvas or some existing canvas mock? I would prefer not to have to ensure that this API is correct (relative to official Canvas API).